lms/djangoapps/learner_dashboard/views.py

```diff
@@ -21,28 +22,33 @@
 @require_GET
 def view_programs(request):
     """View programs in which the user is engaged."""
-    show_program_listing = ProgramsApiConfig.current().show_program_listing
-    if not show_program_listing:
+    programs_config = ProgramsApiConfig.current()
+    if not programs_config.show_program_listing:
         raise Http404
 
     meter = utils.ProgramProgressMeter(request.user)
     programs = meter.engaged_programs
 
     # TODO: Pull 'xseries' string from configuration model.
-    marketing_root = urljoin(settings.MKTG_URLS.get('ROOT'), 'xseries').strip('/')
+    marketing_root = urljoin(settings.MKTG_URLS.get('ROOT'), 'xseries').rstrip('/')
+
     for program in programs:
+        if programs_config.show_program_details:
```

Comment:
Maybe move the block into a new function to keep this view clean and easy to understand?